.circleci/config.yml

```diff
@@ -217,7 +217,34 @@ step-gclient-sync: &step-gclient-sync
           $GCLIENT_EXTRA_ARGS \
           "$CIRCLE_REPOSITORY_URL"
 
-        gclient sync --with_branch_heads --with_tags
+        if ! gclient sync --with_branch_heads --with_tags; then
+          if ELECTRON_USE_THREE_WAY_MERGE_FOR_PATCHES=1 gclient sync -f --with_branch_heads --with_tags; then
+            # Re-export all the new patches now that they've been nicely 3-way
+            # merged
+            python src/electron/script/export_all_patches.py src/electron/patches/config.json
+            # Make a git commit with the updated patches
+            git -C src/electron add patches
+            GIT_COMMITTER_NAME="Electron Bot" GIT_COMMITTER_EMAIL="anonymous@electronjs.org" git commit -m "update patches" --author="Electron Bot <anonymous@electronjs.org>"
+            # Export it
+            mkdir -p patches
+            git -C src/electron am -1 --stdout --keep-subject --no-stat --full-index > patches/update-patches.patch
+            # Upload it to artifacts
+            echo
+            echo "======================================================================"
+            echo "Sync failed with non-3way patch application, but succeeded with 3way."
+            echo "Check the CI artifacts for a patch you can apply to fix it."
+            echo "======================================================================"
+            echo
+          else
+            echo
+            echo "======================================================================"
+            echo "Sync failed even with 3way patch application. You'll need to resolve"
+            echo "this conflict locally."
+            echo "======================================================================"
+            echo
+          fi
+          exit 1
+        fi
```

Comment:
I like this but I'm not sure it belongs in this backport PR?

Reply:
Yeah, I'm still experimenting with it. Once it works I'll move it to another PR. The patch in this PR still doesn't apply.